alibuild_helpers/build.py

```diff
@@ -352,13 +351,26 @@ def doBuild(args, parser):
 
   # Retrieve git heads
   for p in [p for p in buildOrder if "reference" in specs[p]]:
-    cmd = "git ls-remote --heads %s" % specs[p]["reference"]
+    ref_dir = specs[p]["reference"]
+    cmd = None
+
+    if is_writeable(ref_dir):
+      cmd = "git ls-remote --heads %s" % ref_dir
+    else:
+      m = "%s: inexistant or read-only reference dir, will not fetch git heads"
+      debug(m % ref_dir)
+
     if specs[p]["package"] in develPkgs:
       specs[p]["source"] = join(os.getcwd(), specs[p]["package"])
       cmd = "git ls-remote --heads %s" % specs[p]["source"]
-    res, output = getStatusOutputBash(cmd)
-    dieOnError(res, "Error on '%s': %s" % (cmd, output))
-    specs[p]["git_heads"] = output.split("\n")
+
+    output = []
+    if cmd:
+      res, output = getStatusOutputBash(cmd)
```

Comment:
Hmm I would simplify it much more by changing the command, that'd be a one-line modification:

```
cmd = format("touch %(ref)s && git ls-remote --heads %(ref)s || true", ref=specs[p]["source"])
```

This approach has two advantages:

* It actually _tries_ to write instead of trusting the `w` bit (which might not be representative of the actual state, see read-only filesystems, or filesystems which are read-only but can _appear_ to be `rw` like CVMFS or many FUSE modules).
* That'd be a one-line modification.

Concerning actually trying to write, even if it's implemented in Bash, it's more "idiomatic": we try to do something and then we say sorry if we couldn't, instead of checking if we can.

What do you think?

Reply:
yeah, ok :wink:
i will push a commit

Reply:
Actually, do we need the ```|| true```?
Without it, if either the touch or git commands fail,
getBashStatusOutput simply returns a non-zero exit code
along with the error message.